Makefile

```diff
@@ -80,6 +89,84 @@ logs.%: ## Shows the logs of the given docker service. Example: make logs.datast
 restart.%: ## Restarts the given docker service. Example: make restart.datastore
 	docker compose restart $*
 
-run-backend-tests: ## [Not ready for use] Runs the backend tests
-	@echo "Run the backend test on the following module: $(RUN_ARGS)"
-	@echo "Not in use, under construction!"
+run_tests.backend: ## [Not ready for use] Runs the backend tests
+	docker compose up datastore redis -d
+	docker compose run --no-deps --entrypoint "python -m scripts.run_backend_tests $(PYTHON_ARGS)" dev-server
+	$(MAKE) stop
+
+run_tests.frontend: ## Runs the frontend unit tests
+	docker compose run --no-deps --entrypoint "python -m scripts.run_frontend_tests $(PYTHON_ARGS)" dev-server
+
+run_tests.typescript: ## Runs the typescript checks
+	docker compose run --no-deps --entrypoint "python -m scripts.typescript_checks" dev-server
+
+run_tests.custom_eslint: ## Runs the custome eslint tests
+	docker compose run --no-deps --entrypoint "python -m scripts.run_custom_eslint_tests" dev-server
+
+run_tests.mypy: ## Runs mypy checks
+	docker compose run --no-deps --entrypoint "python -m scripts.run_mypy_checks" dev-server
+
+run_tests.backend_associate: ## Runs the backend associate tests
+	docker compose run --no-deps --entrypoint "python -m scripts.check_backend_associated_test_file" dev-server
+
+run_tests.acceptance: ## Runs the acceptance tests for the parsed suite
+	@echo 'Shutting down any previously started server.'
+	$(MAKE) stop 
+	$(MAKE) start-devserver-for-tests
```

Comment:
Add comment why `start-devserver-for-tests` is being used. Do this everywhere, so devs can understand why